src/main/java/sirius/db/mixing/query/QueryCompiler.java

```diff
@@ -520,12 +524,13 @@ protected C compileDefaultSearch(List<QueryField> searchFields, FieldValue token
     /**
      * Permits the subclasses to generate the appropriate search constraint.
      *
-     * @param field the field to search in
-     * @param mode  the mode to use
-     * @param value the value to search for
+     * @param field         the field to search in
+     * @param mode          the mode to use
+     * @param caseSensitive determines if the search should be case-sensitive
+     * @param value         the value to search for
      * @return an appropriate constraint for the given parameters
      */
-    protected abstract C compileSearchToken(Mapping field, QueryField.Mode mode, String value);
+    protected abstract C compileSearchToken(Mapping field, QueryField.Mode mode, boolean caseSensitive, String value);
```

Comment:
This is changing the signature of an existing API, thus making this PR technically BREAKING.